src/JustSaying/AwsTools/MessageHandling/SqsMessagePublisher.cs

```diff
@@ -112,4 +114,121 @@ public InterrogationResult Interrogate()
+            try
+            {
+                response = await _client.SendMessageBatchAsync(request, cancellationToken).ConfigureAwait(false);
+            }
+            catch (AmazonServiceException ex)
+            {
+                throw new PublishBatchException(
+                    $"Failed to publish batch message to SQS. {nameof(request.QueueUrl)}: {request.QueueUrl}",
+                    ex);
+            }
+
+            if (response != null)
+            {
+                using (_logger.BeginScope(new Dictionary<string, string>
+                {
+                    ["AwsRequestId"] = response.ResponseMetadata?.RequestId
+                }))
+                {
+                    if (response.Successful.Count > 0)
+                    {
+                        _logger.LogInformation(
+                            "Published batch of {MessageCount} to {DestinationType} '{MessageDestination}'.",
+                            response.Successful.Count,
+                            "Queue",
+                            request.QueueUrl);
+
+                        foreach (var message in response.Successful)
+                        {
+                            _logger.LogInformation(
+                                "Published message {MessageId} of type {MessageType} to {DestinationType} '{MessageDestination}'.",
+                                message.Id,
+                                message.GetType().FullName,
+                                "Queue",
+                                request.QueueUrl);
+                        }
+                    }
+
+                    if (response.Failed.Count > 0)
+                    {
+                        _logger.LogError(
+                            "Fail to published batch of {MessageCount} to {DestinationType} '{MessageDestination}'.",
+                            response.Failed.Count,
+                            "Queue",
+                            request.QueueUrl);
+
+                        foreach (var message in response.Failed)
+                        {
+                            _logger.LogWarning(
+                                "Fail to published message {MessageId} to {DestinationType} '{MessageDestination}' with error code: {ErrorCode} is error on BatchAPI: {IsBatchAPIError}.",
+                                message.Id,
+                                "Queue",
+                                request.QueueUrl,
+                                message.Code,
+                                message.SenderFault);
+                        }
+                    }
+                }
+            }
+
+            if (MessageBatchResponseLogger != null)
+            {
+                var responseData = new MessageBatchResponse
+                {
+                    SuccessfulMessageIds = response?.Successful.Select(x => x.MessageId),
+                    FailedMessageIds = response?.Failed.Select(x => x.Id),
```

Comment:
As noted in the previous comment, this ensures the queries are materialised once.

```suggestion
                    SuccessfulMessageIds = response?.Successful.Select(x => x.MessageId).ToArray(),
                    FailedMessageIds = response?.Failed.Select(x => x.Id).ToArray(),
```